### ISSUE TITLE
Thunderbird support

### ISSUE THREAD

**Issue by parcelcat** [skhzhang/time-based-themes#29]
This extension works very well in Firefox, and it would be nice to see this extension in Thunderbird, too.

Although #13 mentions that the "System theme" in Thunderbird can switch between light and dark themes, it only works if the operating system has this feature. Not all operating systems have light/dark themes or automatic switching. This extension also offers more options, including theme selection and custom times.

**Comment by Draculah** [comment]
Yes, I would love to see this one in TB either! Although I have other extensions like [gnome night switcher on linux](https://extensions.gnome.org/extension/2236/night-theme-switcher/) which all provide day light support etc, I only need all my applications to change using my keyboard shortcut when I go in lit places. So far, I had to stick with "system theme".

**Comment by skhzhang** [comment]
Hi guys, thanks for the feedback. 

I'll be straight up about this -- I don't use Thunderbird (although maybe I should) and I'm a slave to Gmail at the moment. So I don't have any intention of bringing the extension anytime in the near future.

I'll leave the issue open. It's something I (or someone else) could explore in the "farther" future.

**Comment by Draculah** [comment]
Unfortunate, but I do highly recommend trying out Thunderbird for your email(s), instead of gmail, outlook, ... (as a mail client). It's much faster (offline) and has advanced filtering functionality (just like gmail). It's overwhelming at first though and doesn't look "pretty", but you'll start to appreciate most decisions over time while you find out about all functionality. Yes you'll still be a slave to gmail when using gmail addresses, but less :p 

That way, you can start feeling the need for TB support; heheh ;) I may try out creating extensions for TB/FF myself and porting yours to TB eventually, but I have little experience and for now quite little time.

**Comment by kauron** [comment]
Hi there, it seems that the extension already works fine in Thunderbird 102.6.1, at least the "Manual times" setting.

You can install it manually:
1. Download the .xpi file from [Mozilla's add-ons website](https://addons.mozilla.org/en-US/firefox/addon/automatic-dark/) (right-click the install button and choose "Save link as..."). If you already have it installed, you can create a new profile or access the website from another browser.
2. Open Thunderbird, go into the add-ons manager, click the cogwheel and pick "Install Add-on from file".

Then you can configure exactly like you would on Firefox.

**Comment by Draculah** [comment]
I didn't even test the other options, but I believe the OS based theme listener didn't work with version 91 (as that's what Ubuntu 20.04's apt offered, but well that is updated to 102 because of security issues). I was waiting however for a fix to the OS theme's listener fix (#31), to find out what's happening differently. I didn't know you could install them permanently though, but it makes sense since Firefox allows it as well. Perhaps it's possible to drag and install them as well; thanks!

**Comment by elhennig** [comment]
For me it works well in Thunderbird 115.5.1 on Linux. However, Thunderbird does not open a permission dialog popup to allow the location permissions. It would be great if besides access to location you could also enter Lat-Lon-Values to calculate the sunrise and sunset times, respectively. 
Is that doable at least?

**Comment by Arcticn** [comment]
> Hi guys, thanks for the feedback.
> 
> I'll be straight up about this -- I don't use Thunderbird (although maybe I should) and I'm a slave to Gmail at the moment. So I don't have any intention of bringing the extension anytime in the near future.
> 
> I'll leave the issue open. It's something I (or someone else) could explore in the "farther" future.

Hi, thanks for your work!
This extension can just work fine on ThunderBird, you can remove "location" option from html and directly submit to Thunderbird!
